ui/perfherder/alerts/StatusDropdown.jsx

```diff
@@ -257,38 +258,98 @@ export default class StatusDropdown extends React.Component {
     alertStatus === 'investigating' ||
     (alertStatus !== status && this.isResolved(alertStatus));
 
+  getRemainingTime(firstDate, secondDate) {
+    const differenceInTime = Math.abs(secondDate - firstDate);
+    let differenceInDays = Math.ceil(differenceInTime / (1000 * 60 * 60 * 24));
+
+    // If weekday of date1 is bigger than date2 then it is a new week
+    // and we have to subtract the weekend days
+    if (firstDate.getUTCDay() > secondDate.getUTCDay()) {
+      differenceInDays -= 2;
+    } else {
+      differenceInDays -= 2 * Math.trunc(differenceInDays / 7);
+    }
+
+    let differenceInHours = Math.ceil(differenceInTime / (1000 * 60 * 60));
+    // If due date is Monday and today is Friday and we have to show the hours left,
+    // we have to subtract the weekend hours from the difference
+    if (firstDate.getUTCDay() === 5 && secondDate.getUTCDay() === 1) {
+      differenceInHours -= 2 * 24;
+    }
+
+    return {
+      hours: differenceInHours,
+      days: differenceInDays,
+    };
+  }
+
+  calculateCountdowns(now, dueDate, difference) {
+    if (difference.hours < 24 && difference.hours >= 0) {
+      return `${difference.hours} hours left`;
+    }
+
+    if (now.getTime() >= dueDate.getTime()) {
+      return `Overdue`;
+    }
+
+    return `${difference.days} days left`;
+  }
+
   displayDueDateCountdown() {
+    const { alertSummary } = this.props;
+
     const now = new Date(Date.now());
     const currentDay = now.getDay();
     const saturday = 6;
```

Comment:
These should be moved in a helper/utils file (you can have one specifically for the due date logic, possible names: `dueDateHelper.js`)